src/lib/comms/sol-http-client-impl-curl.c

```diff
@@ -930,6 +962,31 @@ sol_http_client_request(enum sol_http_method method,
     return NULL;
 }
 
+
+SOL_API struct sol_http_client_connection *
+sol_http_client_request(enum sol_http_method method,
+    const char *base_uri, const struct sol_http_params *params,
+    void (*cb)(void *data, const struct sol_http_client_connection *connection,
+    struct sol_http_response *response),
+    const void *data)
+{
+    struct sol_http_client_connection *pending;
+    struct sol_http_request_interface *interface;
+
+    interface = calloc(1, sizeof(*interface));
+    SOL_NULL_CHECK(interface, NULL);
+
+    interface->response_cb = cb;
```

Comment:
here you'd have `struct sol_http_request_interface interface` (no pointer) in the stack, then handle a pointer to it to the internal function.